### PR TITLE
Fix for mobile app suspend/resume

### PR DESCRIPTION
**Pull request by swbsi**
Suspending the app closes the backend RPC/IPC server.

In an app with no other active uv handles (such as setInterval), this releases the last handle holding the uv event loop (internal to node) and the backend exits. On resume, the frontend is then unresponsive.

### CODE REVIEW

**Review by tcobbs-bentley — APPROVED** [review]

File: core/mobile/src/backend/MobileRpcServer.ts [hunk 1]
```diff
@@ -164,6 +164,21 @@ export function setupMobileRpc() {
 
     server = new MobileRpcServer();
     clearInterval(retainUvLoop);
+    retainUvLoop = undefined;
+  });
+
+  MobileHost.onWillTerminate.addListener(() => {
+    if (typeof (retainUvLoop) !== "undefined") {
```
Comment (tcobbs-bentley) [inline core/mobile/src/backend/MobileRpcServer.ts:171]: Unless I'm missing something, this should be:

```ts
if (retainUvLoop !== undefined)
```

File: core/mobile/src/backend/MobileRpcServer.ts [hunk 2]
```diff
@@ -143,6 +143,7 @@ export function setupMobileRpc() {
   }
 
   let server: MobileRpcServer | null = new MobileRpcServer();
+  let retainUvLoop: NodeJS.Timer | undefined;
```
Comment (tcobbs-bentley) [inline core/mobile/src/backend/MobileRpcServer.ts:146]: I think there should be a comment explaining why this exists.

Comment (swbsi) [reply core/mobile/src/backend/MobileRpcServer.ts:146]: Agreed. I'll add one.

File: core/mobile/src/backend/MobileRpcServer.ts (hunk 1 above)
Comment (swbsi) [reply core/mobile/src/backend/MobileRpcServer.ts:171]: I'm pretty sure it has to be that way. There is a weird case where undefined could be a different value. However, typeof is part of the language and is thus more reliable.

**Review by tcobbs-bentley — APPROVED** [review]

**Review by calebmshafer — APPROVED** [review]

**Pull request merged**